src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -60,9 +60,11 @@ private static GitPullRequest CreatePullRequest(string sourceBranch, string targ
             var buildClient = ProjectCollection.GetClient<BuildHttpClient>();
             var definitions = await buildClient.GetDefinitionsAsync(project: Options.TFSProjectName, name: Options.BuildQueueName);
             var builds = await GetBuildsFromTFSAsync(buildClient, definitions, cancellationToken, BuildResult.Succeeded);
+
+            // Get the latest build with valid artifacts.
             return (from build in builds
                     orderby build.FinishTime descending
-                    select build).FirstOrDefault();
+                    select build).FirstOrDefault((b)  => (buildClient.GetArtifactsAsync(b.Project.Id, b.Id, cancellationToken).Result.Any(a => !string.IsNullOrEmpty(a.Name) && a.Name.Contains(b.BuildNumber))));
```

Comment:
It might be safer to do the .Contains() and say something like "contains \\cpvsbuild". I could imagine artifacts ending up somewhere else with the name but we can't consume it that way...?